crates/shared/src/event_handling.rs

```diff
@@ -79,6 +79,10 @@ where
         }
     }
 
+    pub fn store(&self) -> &S {
+        &self.store
+    }
```

Comment:
This is so much nicer than pub crate, but syntactically less concise. Is this a personal preference or are there some other differences coming along with this choice?

If they are the same and the plan is to delete this file in a follow up PR, then I might argue that this change results in more touched lines on an already large PR. 

either way, it's fine.

Reply:
>  Is this a personal preference or are there some other differences coming along with this choice?

There is a subtle difference. Specifically with `pub(crate)` you could do something bad like:
```rust
let mut event_handler = ...;
event_handler.store.change_state_in_unexpected_way();
```

However, if we keep it private and add a public accessor, then you can only get an **immutable** reference to the store, meaning:
```rust
let mut event_handler = ...;
event_handler.store().change_state_in_unexpected_way(); // ERROR: not mutable
```